### ISSUE TITLE
sdkman JAVA_HOME should be used by toolchain if it's the same JDK on OSX

### ISSUE THREAD

**Issue by kenyee** [gradle/gradle#24521]
### Expected Behavior
If SDKMan is used to install JDKs, the Gradle Toolchain should use the same JDK if the JAVA_HOME matches.

### Current Behavior
Gradle toolchain uses the Content/Home location of the JDK on OSX so this appears to be a bug on OSX.
This is because SDKMan on OSX symlinks /Users/keny/.sdkman/candidates/java to the appropriate Java version.

### Context
This causes multiple Gradle daemons to be spawned because the JDK dir doesn't match.

### Steps to Reproduce 
* use SDKMan to install a JDK
* specify a toolchain in Gradle
* add gradle doctor plugin to your gradle file
* run a build on the command line and you'll see this msg:
```
| Gradle is not using JAVA_HOME.                                                                       |
| JAVA_HOME is /Users/keny/.sdkman/candidates/java/11.0.18-zulu                                        |
| Gradle is using /Users/keny/.sdkman/candidates/java/11.0.18-zulu/zulu-11.jdk/Contents/Home

```

### Your Environment
OSX Ventura
Gradle 7.5
SDKman with JDK 11.0.18-zulu  installed

**Comment by ov7a** [comment]
Sorry that you're having trouble with Gradle!

We appreciate the effort that went into filing this issue, but we must ask for more information. 

As stated in our issue template, a [minimal reproducible example](https://en.wikipedia.org/wiki/Minimal_reproducible_example) is a must for us to be able to efficiently track down and fix your problem. Our available resources are severely limited and we must be sure we are looking at the exact problem you are facing.  

If we have a reproducer we may be able to also suggest workarounds or ways to avoid the problem.

The ideal way to provide a reproducer is to leverage our [reproducer template](https://github.com/gradle/gradle-issue-reproducer).

This issue will be closed after 7 days unless you provide more information.

***

Related to https://github.com/gradle/gradle/issues/12840

**Comment by kenyee** [comment]
[gradlesdkman.zip](https://github.com/gradle/gradle/files/11158934/gradlesdkman.zip)

Attached is repro project.
If you run ./gradlew tasks, you'll see this message:
```
  | Gradle is not using JAVA_HOME.                                                                       |
  | JAVA_HOME is /Users/keny/.sdkman/candidates/java/current                                             |
  | Gradle is using /Users/keny/.sdkman/candidates/java/11.0.18-zulu/zulu-11.jdk/Contents/Home           |
  | This can slow down your build significantly when switching from Android Studio to the terminal.      |
  | To fix: Project Structure -> JDK Location.                                                           |
  | Set this to your JAVA_HOME.
```

Hmm... same problem as #12840.  Thanks.
I switched to 22.3.r17-grl/graalvm in sdkman and the problem went away.
TLDR is don't use Zulu w/ Gradle apparently 🤔

**Comment by ov7a** [comment]
So, the core issue here is that this installation uses a lot of symlinks. Everything in JAVA_HOME points to `zulu-11.jdk/Contents/Home`:
```
ls -lF /Users/ov7a/.sdkman/candidates/java/11.0.16-zulu  
total 0
lrwxr-xr-x  1 ov7a  staff  36 Apr  5 19:36 DISCLAIMER@ -> zulu-11.jdk/Contents/Home/DISCLAIMER
lrwxr-xr-x  1 ov7a  staff  38 Apr  5 19:36 Welcome.html@ -> zulu-11.jdk/Contents/Home/Welcome.html
lrwxr-xr-x  1 ov7a  staff  29 Apr  5 19:36 bin@ -> zulu-11.jdk/Contents/Home/bin
lrwxr-xr-x  1 ov7a  staff  30 Apr  5 19:36 conf@ -> zulu-11.jdk/Contents/Home/conf
lrwxr-xr-x  1 ov7a  staff  30 Apr  5 19:36 demo@ -> zulu-11.jdk/Contents/Home/demo
lrwxr-xr-x  1 ov7a  staff  33 Apr  5 19:36 include@ -> zulu-11.jdk/Contents/Home/include
lrwxr-xr-x  1 ov7a  staff  31 Apr  5 19:36 jmods@ -> zulu-11.jdk/Contents/Home/jmods
lrwxr-xr-x  1 ov7a  staff  31 Apr  5 19:36 legal@ -> zulu-11.jdk/Contents/Home/legal
lrwxr-xr-x  1 ov7a  staff  29 Apr  5 19:36 lib@ -> zulu-11.jdk/Contents/Home/lib
lrwxr-xr-x  1 ov7a  staff  29 Apr  5 19:36 man@ -> zulu-11.jdk/Contents/Home/man
lrwxr-xr-x  1 ov7a  staff  36 Apr  5 19:36 readme.txt@ -> zulu-11.jdk/Contents/Home/readme.txt
lrwxr-xr-x  1 ov7a  staff  33 Apr  5 19:36 release@ -> zulu-11.jdk/Contents/Home/release
```

See a related issue:
* https://github.com/gradle/gradle/issues/11200

and a [comment](https://github.com/gradle/gradle/issues/11200#issuecomment-548954595). Please use workaround from [this PR](https://github.com/gradle/gradle/pull/16950#issue-865471613)

**Comment by kenyee** [comment]
I'm not sure how the workaround would work...the tooling (Android Studio/IntelliJ and Gradle are both getting confused with this JDK).  Unless you mean Gradle Doctor is flagging this incorrectly (but I can repro by running ./gradlew on the command line and see multiple daemons started since we're using a Gradle toolchain).

I think the only valid workaround seems to be to use a different JDK since the fixes in that old PR isn't working on OSX.

**Comment by ov7a** [comment]
Did you try to use `/Users/keny/.sdkman/candidates/java/11.0.18-zulu/zulu-11.jdk/Contents/Home` instead of `/Users/keny/.sdkman/candidates/java/11.0.18-zulu ` as your `JAVA_HOME` ?

**Comment by kenyee** [comment]
Yep, that works fine...but sdkman manages JAVA_HOME automagically (probably the main reason to use it)...and all those crazy symlinks.

**Comment by eboudrant** [comment]
@ov7a something not clear for me, this is my error:
```
JAVA_HOME is /Users/eboudrant/.sdkman/candidates/java/11.0.18-zulu                                   
Gradle is using /Users/eboudrant/.sdkman/candidates/java/11.0.18-zulu/zulu-11.jdk/Contents/Home  
```

My `JAVA_HOME` is `/Users/eboudrant/.sdkman/candidates/java/11.0.18-zulu`

Why Gradle append `Contents/Home`? Could you elaborate a bit more on how it work because, as I understand it here, the issue seems to be in in Gradle, not respecting the `JAVA_HOME` set.

**Comment by ov7a** [comment]
@eboudrant Note that this is an output from Gradle Doctor.

Gradle uses `java.home` system property, it might differ from JAVA_HOME (see [thread](https://github.com/gradle/gradle/issues/11200#issuecomment-549012294)).
 
See also https://github.com/runningcode/gradle-doctor/issues/187
